mri/utils/process_SWI

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+import numpy as np
+
+
+def virtual_coil_reconstruction(imgs):
+    """
+    Calculate the combination of all coils using virtual coil reconstruction
+
+    Parameters
+    ----------
+    imgs: np.ndarray
+        The images reconstructed channel by channel
+        in shape [Nch, Nx, Ny, Nz] for 3D or [Nch, Nx, Ny] for 2D
+
+    Returns
+    -------
+    img_comb: np.ndarray
+        The combination of all the channels in a complex valued
+        in shape [Nx, Ny, Nz] for 3D or [Nx, Ny] for 2D
+    """
+    img_sh = imgs.shape
+    # Compute first the virtual coi
+    weights = np.sum(np.abs(imgs), axis=0)
+    weights[weights == 0] = 1e-16
+    phase_reference = np.asarray([np.angle(np.sum(
+        imgs[ch].flatten())) for ch in range(img_sh[0])])
+    reference = np.asarray([(imgs[ch] / weights) / np.exp(
+        1j * phase_reference[ch]) for ch in range(32)])
```

Comment:
`for ch in range(32)` ?
Seems to be hardcoded here, we may need to pickup number of channels from input

Reply:
Also, I think we can still prevent loop here, lemme update here after checking something myself.